python/ray/dag/tests/experimental/test_torch_tensor_dag.py

```diff
@@ -1497,9 +1538,89 @@ def get_transport_name(self) -> str:
 
 
 @pytest.mark.parametrize("ray_start_regular", [{"num_cpus": 4}], indirect=True)
-def test_torch_tensor_nccl_all_reduce_scheduling(ray_start_regular):
+@pytest.mark.parametrize(
+    "operation, op",
+    [
+        (collective.allgather, AllGatherOp()),
+        (collective.allreduce, AllReduceOp()),
+        (collective.reducescatter, ReduceScatterOp()),
+    ],
+)
+def test_torch_tensor_nccl_collective_ops(ray_start_regular, operation, op):
+    """
+    Test basic collective operations.
+    """
+    if not USE_GPU:
+        pytest.skip("NCCL tests require GPUs")
+
+    assert (
+        sum(node["Resources"].get("GPU", 0) for node in ray.nodes()) > 1
+    ), "This test requires at least 2 GPUs"
+
+    actor_cls = TorchTensorWorker.options(num_cpus=0, num_gpus=1)
+
+    num_workers = 2
+    workers = [actor_cls.remote() for _ in range(num_workers)]
+
+    with InputNode() as inp:
+        computes = [
+            worker.compute_with_tuple_args.bind(inp, i)
+            for i, worker in enumerate(workers)
+        ]
+        collectives = operation.bind(computes, op)
+        recvs = [
+            worker.recv_tensor.bind(collective)
+            for worker, collective in zip(workers, collectives)
+        ]
+        dag = MultiOutputNode(recvs)
+
+    compiled_dag = dag.experimental_compile()
+
+    for i in range(3):
+        i += 1
+        shape = (num_workers * i, i)
+        dtype = torch.float16
+        ref = compiled_dag.execute(
+            [(shape, dtype, i + idx) for idx in range(num_workers)]
+        )
+        result = ray.get(ref)
+
+        if operation == collective.allgather:
+            expected_tensor_val = torch.cat(
+                [
+                    torch.ones(shape, dtype=dtype) * i + idx
+                    for idx in range(num_workers)
+                ],
+                dim=0,
+            )
+        elif operation == collective.allreduce:
+            expected_tensor_val = torch.ones(shape, dtype=dtype) * sum(
+                i + idx for idx in range(num_workers)
+            )
+        elif operation == collective.reducescatter:
+            expected_tensor_val = torch.ones((i, i), dtype=dtype) * sum(
+                i + idx for idx in range(num_workers)
+            )
+        else:
+            raise ValueError(f"Unknown operation: {operation}")
+
+        for tensor in result:
+            tensor = tensor.to("cpu")
+            assert torch.equal(tensor, expected_tensor_val)
+
+
+@pytest.mark.parametrize("ray_start_regular", [{"num_cpus": 4}], indirect=True)
+@pytest.mark.parametrize(
+    "operation, op",
+    [
+        (collective.allgather, AllGatherOp()),
+        (collective.allreduce, AllReduceOp()),
+        (collective.reducescatter, ReduceScatterOp()),
+    ],
+)
```

Comment:
It's okay to just use one collective op for the remaining tests. We only need one to check for the correctness of the output values.